### ISSUE TITLE
Support multiple instances of Excalidraw

### ISSUE THREAD

**Issue by ad1992** [excalidraw/excalidraw#3043]
Tasklist to support [multiple instances of Excalidraw](https://k1xx5.csb.app/)

- [x] keyboard Events should be attached to only the current instance, right now it applies to all instances.(fixed in https://github.com/excalidraw/excalidraw/pull/3430)
- [x] History is currently shared, It should be separate for each instance (fixed in https://github.com/excalidraw/excalidraw/pull/3481)
- [x] Adding text has offsets problem as mentioned in https://github.com/excalidraw/excalidraw/issues/3043#issuecomment-792237193 , fixed in https://github.com/excalidraw/excalidraw/pull/3478
- [x] Fix Dark mode as explained in https://github.com/excalidraw/excalidraw/issues/3043#issuecomment-792237891 (fixed in https://github.com/excalidraw/excalidraw/pull/3446)
- [x] Library is shared (fixed in #3451)

...what more?

**Comment by zsviczian** [comment]
Multiple instances of Excalidraw on the same page, add text issues on some of the instances
![New video](https://user-images.githubusercontent.com/14358394/110233663-aabd9200-7f25-11eb-837b-b45b69108d9c.gif)

**Comment by zsviczian** [comment]
Similar issue with dark mode / light mode:
![2021-03-07 09-22-14](https://user-images.githubusercontent.com/14358394/110233827-03415f00-7f27-11eb-80df-eff0cfc6e4ba.gif)

**Comment by ad1992** [comment]
closing this as all the issues related to multiple excalidraw instances are fixed and will be available in the next release 🎉 . Please open a new issue for further issues.

**Comment by gaearon** [comment]
Hey, I just wanted to really thank you for working on this. It's been a huge help in my personal project to see this fixed because it unblocked a bunch of things I couldn't do before and corrected many bugs. Appreciated.

**Comment by ad1992** [comment]
> Hey, I just wanted to really thank you for working on this. It's been a huge help in my personal project to see this fixed because it unblocked a bunch of things I couldn't do before and corrected many bugs. Appreciated.

Thank you for the kind words, super glad to know that 🎉
